### PR TITLE
RAB-998: Add service API to delete user

### PR DESCRIPTION
**Pull request by pierrejolly**
**Description (for Contributor and Core Developer)**

In this PR I added a new service API to delete an user

### CODE REVIEW

File: src/Akeneo/UserManagement/back/ServiceApi/User/DeleteUserCommand.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+namespace Akeneo\UserManagement\ServiceApi\User;
+
+final class DeleteUserCommand
+{
+    private function __construct(
+        public string $username,
+    ) {
+    }
+
+    public static function delete(
```
Comment (ghoussard) [inline src/Akeneo/UserManagement/back/ServiceApi/User/DeleteUserCommand.php:19]: This static method is a bit useless, we can use the constructor, there is only one parameter 🤔

File: src/Akeneo/UserManagement/back/tests/Integration/ServiceApi/User/DeleteUserHandlerIntegration.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+use Akeneo\Test\Integration\Configuration;
+use Akeneo\Test\Integration\TestCase;
+use Akeneo\UserManagement\Component\Model\UserInterface;
+use Akeneo\UserManagement\ServiceApi\User\DeleteUserCommand;
+use Akeneo\UserManagement\ServiceApi\User\DeleteUserHandlerInterface;
+use Symfony\Component\Security\Core\Exception\UserNotFoundException;
+
+class DeleteUserHandlerIntegration  extends TestCase
+{
+    public function testItDeleteAUser()
```
Comment (ghoussard) [inline src/Akeneo/UserManagement/back/tests/Integration/ServiceApi/User/DeleteUserHandlerIntegration.php:22]:
```suggestion
    public function testItDeleteAUser(): void
```

File: src/Akeneo/UserManagement/back/tests/Integration/ServiceApi/User/DeleteUserHandlerIntegration.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    {
+        $username = 'a_user';
+
+        $this->createUser($username);
+        $this->assertUserExist($username);
+        $this->deleteUser($username);
+        $this->assertUserDoesNotExist($username);
+    }
+
+    public function testItThrowsAnExceptionUserDoesNotExist()
```
Comment (ghoussard) [inline src/Akeneo/UserManagement/back/tests/Integration/ServiceApi/User/DeleteUserHandlerIntegration.php:32]:
```suggestion
    public function testItThrowsAnExceptionUserDoesNotExist(): void
```

**Review by ghoussard — APPROVED** [review]

File: src/Akeneo/UserManagement/back/Application/Handler/DeleteUserHandler.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    public function handle(DeleteUserCommand $deleteUserCommand): void
+    {
+        /** @var UserInterface|null $user */
+        $user = $this->userRepository->findOneBy(['username' => $deleteUserCommand->username]);
+
+        if (null === $user) {
+            throw new UserNotFoundException();
+        }
+
+        $this->userManager->deleteUser($user);
```
Comment (grena) [inline src/Akeneo/UserManagement/back/Application/Handler/DeleteUserHandler.php:36]: I wonder if we should make the `UserManager` handle every Doctrine/MySQL related exceptions to throw real domain exceptions, but maybe not in this PR.

**Review by grena — APPROVED** [review]

**Pull request merged**